earthaccess/search.py

```diff
@@ -640,10 +641,15 @@ def granule_name(self, granule_name: str) -> Self:
             self
 
         Raises:
-            TypeError: if `granule_name` is not of type `str`
+            TypeError: if `granule_name` is not of type `str` or `Iterable[str]`.
         """
+        if not isinstance(granule_name, Iterable):
+            raise TypeError("granule_name must be of type string or Iterable of strings")
         if not isinstance(granule_name, str):
-            raise TypeError("granule_name must be of type string")
+            # Convert iterable to list of strings. Since str is also Iterable, make
+            # sure we don't do this when granule_name is a string, otherwise
+            # we would get a list of individual characters.
+            granule_name = list(str(name) for name in granule_name)
```

Comment:
Sorry, I should have simply suggested using a list comprehension earlier:

```suggestion
            granule_name = [str(name) for name in granule_name]
```